### PR TITLE
[KOA-4588] Add XL icon tests and examples

### PR DESCRIPTION
**Pull request by georgegillams**
Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ NA `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

File: Backpack/Icon/Classes/BPKXlIconView.m [hunk 1]
```diff
@@ -25,6 +25,17 @@ - (instancetype)initWithIconName:(nullable BPKXlIconName)iconName {
     return [super initWithIconName:iconName size:BPKIconSizeXLarge];
 }
 
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
```
Comment (georgegillams) [inline Backpack/Icon/Classes/BPKXlIconView.m:30]: This was missing from the XLIconView, but was present in the other 2 classes. As a result the icons were being rendered too small in the example app.

**Review by olliecurtis — APPROVED** [review]

File: Example/Backpack/Views/CollectionView Cells/IconsPreviewCollectionViewCell.swift [hunk 2]
```diff
@@ -79,9 +90,7 @@ class IconsPreviewCollectionViewCell<T: IconDetails>: UICollectionViewCell {
 
         NSLayoutConstraint.activate([
             imageView.centerXAnchor.constraint(equalTo: contentView.centerXAnchor),
-
-            imageView.topAnchor.constraint(greaterThanOrEqualTo: contentView.topAnchor, constant: BPKSpacingSm),
-            contentView.bottomAnchor.constraint(greaterThanOrEqualTo: imageView.bottomAnchor, constant: BPKSpacingSm)
```
Comment (georgegillams) [inline Example/Backpack/Views/CollectionView Cells/IconsPreviewCollectionViewCell.swift:84]: These were just centering on the Y axis, so I made it more readable.

File: Backpack/Icon/Classes/BPKXlIconView.m (hunk 1 above)
Comment (shaundon) [reply Backpack/Icon/Classes/BPKXlIconView.m:30]: Maybe it should be a protocol to enforce that it exists? Is that even a thing in Obj-C or is it just Swift?

File: UNRELEASED.md [hunk 3]
```diff
@@ -3,8 +3,12 @@
 > Place your changes below this line.
 
 **Fixed:**
+
 - Migrated to new Backpack Foundations packages, `@skyscanner/bpk-ios` and `@skyscanner/bpk-svgs`.
 
+- Backpack/Icon
+ - Minor changes to `BPKIcon` which should not affect how the are used or work.
```
Comment (shaundon) [inline UNRELEASED.md:10]:
```suggestion
 - Minor changes to `BPKIcon` which should not affect how they are used or work.
```

**Comment by georgegillams** [comment]
@shaundon 
> Maybe it should be a protocol to enforce that it exists? Is that even a thing in Obj-C or is it just Swift?

I'm not sure if that'd help as `initWithFrame` is defined by the superclass. The thing we need to ensure is that we've overridden it and set the `size` property correctly.

I think the tests I've added to verify it works as expected are sufficient. Might be more we can do though?

**Review by shaundon — APPROVED** [review]

**Pull request merged**